pkg/util/pem/pem.go

```diff
@@ -8,8 +8,13 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
+	"io"
 )
 
+type Encodable interface {
```

Comment:
Just curious - why is this interface needed? It looks like you already have logic handling these cert types in your switch/case on lines 64-83.

Reply:
It's needed to define the type constraint on the generic parameter for the function. 

FWIW this implementation was largely lifted from our equivalent code in ARO-RP: https://github.com/Azure/ARO-RP/blob/master/pkg/util/pem/encode.go. I just changed this to output directly to an `io.Writer` rather than assembling the contents in an in-memory `[]byte`.